### PR TITLE
Add native Tramp VTX support

### PR DESCRIPTION
**Pull request by MUSTARDTIGERFPV**
## Description

This PR adds a native Tramp protocol VTX as a Serial1 protocol. After selecting Tramp as your Serial1 protocol, any VTX Admin actions will send commands to a VTX attached to Serial1's TX pin.

### Bidirectionality

Responses from the VTX aren't interpreted at all. This is one-way "set the VTX" support only.

### Power levels
Tramp doesn't define "power levels", while our VTX Admin setup does. To that end, I had to determine my own LUT for power levels unless we want to have full VTX tables in ExpressLRS which _ehhhhhhhhhhhhhhhhhhh_ no.

| Power Level | Output Power |
|-------------|--------------|
| 1           | 10mW          |
| 2           | 25mW          |
| 3           | 200mW        |
| 4           | 400mW        |
| 5           | 600mW        |
| 6           | 1000mW      |
| 7           | 1600mW      |
| 8           | 3000mW      |

### 1.3GHz/2.4GHz support

Since we only have A/B/E/F/R/L bands, 1.3GHz/2.4GHz VTXes aren't supported by this. It could be theoretically added, but at this time it's not a priority to support.

## Testing

Tested on OpenVTX

### CODE REVIEW

File: src/src/rx-serial/SerialTramp.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+void SerialTramp::sendQueuedData(uint32_t maxBytesToSend)
+{
+    uint32_t bytesWritten = 0;
+    while (_fifo.size() > 0 && bytesWritten < maxBytesToSend) {
+        _fifo.lock();
+        uint8_t frameSize = _fifo.pop() - 1;
+        uint8_t frame[frameSize];
+        _fifo.popBytes(frame, frameSize);
```
Comment (wvarty) [inline src/src/rx-serial/SerialTramp.cpp:24]: Should we be locking and unlocking the `_fifo` for all these read operations, so that we are using the compile flags to protect the queue during r/w?

Comment (wvarty) [reply src/src/rx-serial/SerialTramp.cpp:24]: Oh we are lol. We do it higher up. Can we move the lock right be fore the popBytes so that this doesn't catch the next person out?

Comment (wvarty) [reply src/src/rx-serial/SerialTramp.cpp:24]: Hmm nevermind. there's a rogue `pop` in there that also gets covered by the lock. Ignore me.

**Review by wvarty — APPROVED** [review]

File: src/user_defines.txt
Comment (pkendall64) [inline src/user_defines.txt:1]: Changes to this file should not be part of the PR!

**Review by pkendall64 — APPROVED** [review]

File: src/src/rx-serial/SerialTramp.cpp [hunk 2]
```diff
@@ -42,8 +43,11 @@ void SerialTramp::sendQueuedData(uint32_t maxBytesToSend)
         _fifo.unlock();
         setTXMode();
         _outputPort->write(frame, frameSize);
-        setRXMode();
         bytesWritten += frameSize;
+        lastSendTime = millis();
+    }
+    if (_fifo.size() == 0 && millis() - lastSendTime > 100) {
```
Comment (pkendall64) [inline src/src/rx-serial/SerialTramp.cpp:49]: Rather than using a time based check you could check if the TX is idle
```cpp
        if (_fifo.size() == 0 && uart_ll_is_tx_idle(UART_LL_GET_HW(1)))
```

Comment (MUSTARDTIGERFPV) [reply src/src/rx-serial/SerialTramp.cpp:49]: Oh that's so much better thank you

**Review by JyeSmith — APPROVED** [review]
